jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -2282,9 +2282,9 @@ public void setSNIProvider(SniProvider sniProvider)
                 String host = sslEngine.getPeerHost();
                 if (host != null)
                 {
-                    // TODO Must handle : somehow as java17 SNIHostName never handles:  See #6624
                     // Must use the byte[] constructor, because the character ':' is forbidden when
                     // using the String constructor (but typically present in IPv6 addresses).
+                    // Since Java 17, only letter|digit|hyphen characters are allowed, even by the byte[] constructor.
```

Comment:
This is still not a proper fix.   If `:` is never allowed, then why are we using the byte[] version?  Shouldn't we just revert to the string version and remove this comment?